### PR TITLE
Run Diamond core tests

### PR DESCRIPTION
**Pull request by gnufied**
This just enables running diamond core tests as part of travis build process.

I am still tweaking this for how best to run python tests in different environments etc and still keep fullerite build cycle relatively fast.

### CODE REVIEW

**Comment by zachm** [comment]
+1 given that this works in all environments. @gnufied, what do you feel is a reasonable additional test runtime? And how close is this PR to achieving that?

**Comment by gnufied** [comment]
Currently the diamond core tests take:

```
Ran 9 tests in 0.002s
```

to run, but I am thinking of keeping diamond core or collector tests out  from `make` default command which we often use locally to test fullerite core or build new package locally. 

So in Jenkins or Travis, both diamond core and diamond collector tests are ran separately but if a developer just runs `make` command locally, it won't run diamond core test or diamond collector test. What you guys think? cc @tsheasha @venkey-ariv  @zachm 

Also, the core tests are running as you can see, the reason I marked this PR as WIP is - I was thinking of adding some tests for dimension support in diamond we have added, but I am leaning towards doing that in separate request now.

Also, lot of collector tests are still failing but that is being dealt in separate ticket.

**Comment by tsheasha** [comment]
1) The test run this quick because they are basically empty

2) keeping them out of **make** does not seem right to me =, if we want to avoid tests for dev envs we can add a command **make dev**.

3) The core tests are running and passing because they are not doing basically much, so if you keep this PR as it is and not add tests for our dimensions then I would just say close this PR and when we do add tests for our changes we can create a new PR, otherwise this just adds really no-op code. IMO

**Comment by gnufied** [comment]
1. Yeah unfortunately diamond's own test suite is pretty bare when it comes to diamond core tests - https://github.com/python-diamond/Diamond/tree/master/src/diamond/test 

2. Yeah I can add some tests for some of the stuff we have added to diamond core, no worries to make this PR worthwhile. 

3. `make` vs `make dev` - no strong opinions really, I was just thinking people are more used to `make` running quickly and now it will come slow and hence my reasoning.

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 language: go
 
+python:
+  - "2.6"
```
Comment (poros) [inline .travis.yml]: very minor: I guess you don't need double quotes

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+configobj
```
Comment (poros) [inline requirements.txt]: Can you call this file requirements-dev.txt?
These are not really requirements for production environments

File: .travis.yml (hunk 1 above)
Comment (gnufied) [inline .travis.yml]: Taken from - https://docs.travis-ci.com/user/languages/python . I don't think it is working though at all though. multi-language projects are bit of problem with travis.

File: requirements.txt (hunk 2 above)
Comment (gnufied) [inline requirements.txt]: good idea.

**Comment by tsheasha** [comment]
+1

**Comment by venkey-ariv** [comment]
+1

**Pull request merged**